mobile-app/app/screens/TransactionAuthorization/PasscodePrompt.tsx

```diff
@@ -58,91 +59,126 @@ export function PasscodePrompt (props: PasscodePromptProps): JSX.Element {
+                    }}
+                    testID='pin_authorize'
+                    value={props.pin}
+                  />
+                )
+              }
+
+              <Loading
+                message={props.status === 'SIGNING' ? props.loadingMessage : undefined}
+              />
+
+              {// upon retry: show remaining attempt allowed
+              (props.isRetry && props.attemptsRemaining !== undefined && props.attemptsRemaining !== props.maxPasscodeAttempt)
+                ? (
+                  <ThemedText
+                    dark={tailwind('text-darkerror-500')}
+                    light={tailwind('text-error-500')}
+                    style={tailwind('text-center text-sm font-bold mt-5')}
+                    testID='pin_attempt_error'
+                  >
+                    {translate('screens/PinConfirmation', `${props.attemptsRemaining === 1
+                      ? 'Last attempt or your wallet will be unlinked for your security'
+                      : 'Incorrect passcode. {{attemptsRemaining}} attempts remaining'}`, { attemptsRemaining: props.attemptsRemaining })}
+                  </ThemedText>
+                )
+                : null
+            }
+
+              {// on first time: warn user there were accumulated error attempt counter
+              (!props.isRetry && props.attemptsRemaining !== undefined && props.attemptsRemaining !== props.maxPasscodeAttempt)
+                ? (
+                  <ThemedText
+                    dark={tailwind('text-darkerror-500')}
+                    light={tailwind('text-error-500')}
+                    style={tailwind('text-center text-sm font-bold mt-5')}
+                    testID='pin_attempt_warning'
+                  >
+                    {translate('screens/PinConfirmation', `${props.attemptsRemaining === 1
+                      ? 'Last attempt or your wallet will be unlinked for your security'
+                      : '{{attemptsRemaining}} attempts remaining'}`, { attemptsRemaining: props.attemptsRemaining })}
+                  </ThemedText>
+                )
+                : null
+            }
+            </ThemedView>)}
       </ThemedView>
     </SafeAreaView>
   )
 }
 
+function SuccessMessage ({ message }: { message?: {title: string, description: string}}): JSX.Element | null {
+  if (message === undefined) {
+    return null
+  }
+
+  return (
+    <View style={tailwind('flex-col items-center p-6')}>
+      <ThemedIcon
+        dark={tailwind('text-darksuccess-500')}
+        iconType='MaterialIcons'
+        light={tailwind('text-success-500')}
+        name='check-circle'
+        size={26}
+      />
+      <ThemedText style={tailwind('text-center text-xl font-bold mt-5')}>
+        {message.title}
+      </ThemedText>
+
+      <ThemedText style={tailwind('ml-2 mt-2')}>
+        {message.description}
+      </ThemedText>
+    </View>
+  )
+}
+
 function Loading ({ message }: { message?: string }): JSX.Element | null {
```

Comment:
Can you adjust the size of the loading message also? Currently I think it's using base font size

Reply:
> You also need to check the 2nd case when the passcode prompt is displayed. The 2nd case is displayed when you're decrypting your mnemonic. You can check it via Settings => View Recovery words. We should also have the completion screen displayed. You can check with @jtlntno-cake for an updated copy.

Yes I noticed that, thought it was intentional! 🙈  @jtlntno-cake will comment in https://github.com/DeFiCh/wallet/issues/963 once it's ready!